### PR TITLE
Permissions: Unify Location

### PR DESCRIPTION
**Pull request by malmstein**
Task/Issue URL: https://app.asana.com/0/1174433894299346/1208719691317744/f

### Description
This PR moves the Location permission logic to the SitePermissionsManager
Note: This 

### Steps to test this PR

_Location permission granted_
- [x] Fresh install and visit permission.site
- [x] Ask for Location permissions
- [x] Verify new Location dialog appears
- [x] Allow permissions
- [x] Verify system dialog is shown
- [x] Allow permissions
- [x] Verify location is granted (location button is green)

_Location permission not granted_
- [x] Fresh install and visit permission.site
- [x] Ask for Location permissions
- [x] Verify new Location dialog appears
- [x] Deny permissions
- [x] Verify location is not granted (location button is red)

_Location permission granted (not system granted)_
- [x] Fresh install and visit permission.site
- [x] Ask for Location permissions
- [x] Verify new Location dialog appears
- [x] Allow permissions
- [x] Verify system dialog is shown
- [x] Deny permissions
- [x] Verify location is not granted (location button is red)
- [x] Verify Snackbar appears
- [x] Ask for Location permissions
- [x] Verify system dialog is shown
- [x] Deny permissions
- [x] Verify Settings dialog appears
- [x] Tap on Open Settings
- [x] Verify Device Settings screen opens

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208719691317744

### CODE REVIEW

**Comment by malmstein** [comment]
* **#5208** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5208?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @malmstein and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: app/src/androidTest/java/com/duckduckgo/app/location/GeoLocationPermissionsTestRequest.kt [hunk 1]
```diff
@@ -39,7 +39,7 @@ import org.junit.Rule
 import org.junit.Test
 import org.mockito.kotlin.mock
 
-class GeoLocationPermissionsTest {
+class GeoLocationPermissionsTestRequest {
```
Comment (anikiki) [inline app/src/androidTest/java/com/duckduckgo/app/location/GeoLocationPermissionsTestRequest.kt:42]: Could this be `GeoLocationPermissionsRequestTest` (ending in test)?

File: app/src/androidTest/java/com/duckduckgo/app/location/data/LocationPermissionsRepositoryImplTestRequest.kt [hunk 2]
```diff
@@ -35,7 +35,7 @@ import org.mockito.MockitoAnnotations
 import org.mockito.kotlin.mock
 import org.mockito.kotlin.verify
 
-class LocationPermissionsRepositoryImplTest {
+class LocationPermissionsRepositoryImplTestRequest {
```
Comment (anikiki) [inline app/src/androidTest/java/com/duckduckgo/app/location/data/LocationPermissionsRepositoryImplTestRequest.kt:38]: Same as above (end in "Test").

File: app/src/main/java/com/duckduckgo/app/browser/BrowserChromeClient.kt [hunk 3]
```diff
@@ -144,16 +145,28 @@ class BrowserChromeClient @Inject constructor(
...
     }
 
+    override fun onGeolocationPermissionsShowPrompt(
+        origin: String,
+        callback: GeolocationPermissions.Callback,
+    ) {
+        Timber.d("Permissions: location permission requested $origin")
+        onPermissionRequest(LocationPermissionRequest(origin, callback))
+
+        // webViewClientListener?.onSiteLocationPermissionRequested(origin, callback)
```
Comment (anikiki) [inline app/src/main/java/com/duckduckgo/app/browser/BrowserChromeClient.kt:167]: Note to remove commented out line.

File: app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt [hunk 4]
```diff
@@ -1975,52 +1972,54 @@ class BrowserTabFragment :
...
-            viewModel.onSiteLocationPermissionSelected(domain, LocationPermissionType.DENY_ALWAYS)
-            dialog.dismiss()
-        }
-
-        dialog.show()
+        activity?.let {
+            sitePermissionsDialogLauncher.showSiteLocationPermissionDialog(it, locationPermissionRequest, tabId)
+        }
+
+        // val binding = ContentSiteLocationPermissionDialogBinding.inflate(layoutInflater)
```
Comment (anikiki) [inline app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt:1985]: Note to remove commented out code (I see it was moved).

**Review by anikiki — APPROVED** [review]
Looks good and works as expected! :tada:
Left a few tiny comments.

File: app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt (hunk 4 above)
Comment (malmstein) [reply app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt:1985]: It’s removed a bit higher, alongside the resources, etc

**Review by nalcalag — CHANGES_REQUESTED** [review]
@malmstein I tested your branch and there is a problem when the site requests more than one site permission.
You can reproduce it on https://webcammictest.com/ which would request mic and camera. I checked and it works fine in production.
The bug was introduced in branch `feature/david/11-04-permissions_update_drm_dialogs`

File: app/src/main/java/com/duckduckgo/app/sitepermissions/SitePermissionsViewModel.kt [hunk 5]
```diff
@@ -63,20 +55,17 @@ class SitePermissionsViewModel @Inject constructor(
...
-            val removedLocationPermissions: List<LocationPermissionEntity>,
-        ) : Command()
+        class ShowRemovedAllConfirmationSnackbar(val removedSitePermissions: List<SitePermissionsEntity>) : Command()
         class LaunchWebsiteAllowed(val domain: String) : Command()
     }
 
     init {
         _viewState.value = ViewState(
-            askLocationEnabled = settingsDataStore.appLocationPermission,
+            // askLocationEnabled = settingsDataStore.appLocationPermission,
```
Comment (anikiki) [inline app/src/main/java/com/duckduckgo/app/sitepermissions/SitePermissionsViewModel.kt:67]: Note to remove commented out line.

**Review by anikiki — APPROVED** [review]
LGTM! :+1: 
There's a failing test.

**Review by anikiki — APPROVED** [review]
Checked the last changes. LGTM :+1:

**Pull request merged**